src/Storage/Token.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
```

Comment:
> Could you explain to me what the criteria for grouping things in the Storage namespace is? We need to come up with a better name, IMO.

My idea was to group things related to token structure. Maybe `Builder` and `Parser` doesn't belong there.

> Is there a big difference between encrypted and plain tokens?

In terms of structure, plain tokens have `headers`, `claims` and `signature` (optionally). Encrypted tokens have `headers`, `encrypted key`, `initialization vector`, `ciphertext` and `authentication tag`.

I would say that they should be represented with different objects.